### PR TITLE
Add plugin to support ktor-server-resources

### LINKED ISSUE #356: Add support for ktor Resources in addition to / instead of Locations
**Is your feature request related to a problem? Please describe.**

According to [Locations documentation](https://ktor.io/docs/locations.html), the plugin is deprecated and [Resources](https://ktor.io/docs/type-safe-routing.html) should be used instead:

> In v2.0.0, the Locations plugin is deprecated. Use the [Resources](https://ktor.io/docs/type-safe-routing.html) plugin instead.

**Describe the solution you'd like**

Replace the kompendium-locations plugin with kompendium-resources. Looking at the code, a simple copy-paste of existing plugin code with some renaming and adding support for `Resource` annotations should do the trick. (Can also just add a new plugin in addition to existing locations, but I don't see much point in keeping a plugin for deprecated ktor functionality.)

**Describe alternatives you've considered**

The workaround I'm using now is annotating the resource with both `Location` and `Resource` annotations. It seems to work, but is not a very good option.

### PR DESCRIPTION
**Pull request by serpro69**
# Description

Adds support for ktor resources plugin.

Closes #356 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Unit tests + playground

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have updated the CHANGELOG in the `Unreleased` section
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by serpro69** [comment]
Pending (hence a draft for now): 
- [ ] documentation
- [ ] playground 
- [ ] deprecation warning for locations plugin

 Will add this in the coming days when I find a free hour to kill :)

File: locations/src/main/kotlin/io/bkbn/kompendium/locations/NotarizedLocations.kt [hunk 1]
```diff
@@ -43,7 +43,6 @@ object NotarizedLocations {
     name = "NotarizedLocations",
     createConfiguration = ::Config
   ) {
-    println("hi")
```
Comment (unredundant) [inline locations/src/main/kotlin/io/bkbn/kompendium/locations/NotarizedLocations.kt:46]: 🫡

**Review by unredundant — COMMENTED** [review]
this looks great, once you knock off those items you mentioned in the list above lets ship this :shipit:

**Comment by serpro69** [comment]
All done and ready for review :)

**Comment by unredundant** [comment]
Great work, thanks again!

**Pull request merged**